File: backend/src/utils/bedrockModelId.ts
/**
 * Bedrock cross-region inference profile IDs use a `us.` prefix
 * (e.g. us.anthropic.claude-haiku-4-5-20251001-v1:0).
 */

/** Default / fallback Bedrock model — keep in sync with python/app/env_bootstrap.py */
export const DEFAULT_BEDROCK_MODEL =
  "us.anthropic.claude-3-sonnet-20240229-v1:0:200k";

const BEDROCK_MODEL_ID_PATTERN = /^[a-z0-9][a-z0-9-]*\.[a-z0-9][a-z0-9.:-]*$/i;

/** Pass through configured model id; use default when env is empty. */
export function normalizeBedrockModelAlias(modelId: string): string {
  const trimmed = modelId.trim();
  if (!trimmed) return DEFAULT_BEDROCK_MODEL;
  return trimmed;
}

export function stripUsModelPrefix(modelId: string): string {
  const trimmed = modelId.trim();
  if (trimmed.toLowerCase().startsWith("us.")) {
    return trimmed.slice(3);
  }
  return trimmed;
}

/** True for provider-style Bedrock ids (anthropic.*, meta.*, …), not short aliases. */
export function isBedrockProviderModelId(modelId: string): boolean {
  const trimmed = modelId.trim();
  if (!trimmed || trimmed.includes("/")) return false;
  return BEDROCK_MODEL_ID_PATTERN.test(trimmed);
}

export function withUsModelPrefix(modelId: string): string {
  const trimmed = modelId.trim();
  if (!trimmed) return trimmed;
  if (trimmed.toLowerCase().startsWith("us.")) return trimmed;
  if (isBedrockProviderModelId(trimmed)) {
    return `us.${trimmed}`;
  }
  return trimmed;
}

File: python/app/llm/bedrock_model_id.py
"""Bedrock model id helpers — US cross-region inference profile prefix."""

from __future__ import annotations

import re

_BEDROCK_MODEL_ID_RE = re.compile(
    r"^[a-z0-9][a-z0-9-]*\.[a-z0-9][a-z0-9.:-]*$",
    re.IGNORECASE,
)


def strip_us_model_prefix(model_id: str) -> str:
    trimmed = (model_id or "").strip()
    if trimmed.lower().startswith("us."):
        return trimmed[3:]
    return trimmed


def is_bedrock_provider_model_id(model_id: str) -> bool:
    trimmed = (model_id or "").strip()
    if not trimmed or "/" in trimmed:
        return False
    return bool(_BEDROCK_MODEL_ID_RE.match(trimmed))


def with_us_model_prefix(model_id: str) -> str:
    trimmed = (model_id or "").strip()
    if not trimmed:
        return trimmed
    if trimmed.lower().startswith("us."):
        return trimmed
    if is_bedrock_provider_model_id(trimmed):
        return f"us.{trimmed}"
    return trimmed
